journal/202309.md

```diff
@@ -0,0 +1,971 @@
+- Added support for [CUDA versions older than 10](https://github.com/decred/gominer/pull/216). This allows to build gominer on more configurations, although performance will be significantly degraded with older CUDA. Installed CUDA toolkit version and GPU architecture will be [detected automatically](https://github.com/decred/gominer/pull/n/commits/a7076d78a46f9b9d9277fb3f92620ed130d5e7e9) during build to use the best possible GPU features, specifically the fast [funnel shift](https://stackoverflow.com/questions/12767113/funnel-shift-what-is-it) intrinsic.
+- Fixed an issue where [autocalibration](https://github.com/decred/gominer/pull/218/commits/d3e68cf1e370e94eb50b1e74d5b2446b8e24c4ad) could not release the device and make the actual mining loop fail.
+
+GPU programming is not something the developers do every day but this OpenCL kernel has shown to be quite fast. For example, it achieves 10-12 GH/s on an RTX 3070, which is close to 12 GH/s reported by [WhatToMine](https://whattomine.com/gpus/48-nvidia-geforce-rtx-3070) for IronFish, another mining algorithm based on BLAKE3. A better comparison would need to consider the details (size of the hashed data, number of hashing rounds, etc.), but the hashrates reported so far suggest that gominer is good enough to make GPU mining fairer and accessible to more people.
+
+README updates:
+
+- Switched Windows OpenCL build instructions to [MSYS2](https://github.com/decred/gominer/pull/196).
+- Added OpenCL AMD/Nvidia [build instructions](https://github.com/decred/gominer/pull/198) for Linux.
+- Added [configuration instructions](https://github.com/decred/gominer/pull/199).
+- Added [user-reported hashrates](https://github.com/decred/gominer/pull/201).
+
+Other changes:
+
+- Improved [shutdown handling](https://github.com/decred/gominer/pull/207).
+- 14 [housekeeping](https://github.com/decred/gominer/pull/210) and clean up commits including the addition of more linters to prevent poor code from getting in.
+- Fixed a [couple](https://github.com/decred/gominer/pull/204) of [issues](https://github.com/decred/gominer/pull/205) with the [Stratum](https://braiins.com/stratum-v1) protocol.
+
+{ _omitting minor changes: 202, 209, 211, 213, 215_ }
+
+
+### dcrpool
+
+_[dcrpool](https://github.com/decred/dcrpool) is server software for running a mining pool._
+
+Similar to gominer, dcrpool development has been reactivated after a period of low activity since summer 2021 when release v1.2.0 came out. The developers upgraded the codebase to latest Go features, fixed multiple concurrency and shutdown edge cases, and paid some technical debt, all in preparation of adding pooled GPU mining support.
+
+Upgrades and changes:
+
+- Added [BLAKE3 support](https://github.com/decred/dcrpool/pull/341) to the internal CPU miner which is used for testing.
+  - { _it is for testing right?_ }
```

Comment:
Yes.  It's not production ready at all as, among other things, it only uses a single core with no option to use more.

Reply:
Thanks, updated.